### PR TITLE
fix index.d.ts

### PR DESCRIPTION
**Pull request by Gavinchen92**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -4,10 +4,10 @@ type NextConfiguration = any;
 
 export = withAwesomeTypescript;
 
-declare function withAwesomeTypescript(
-  awesomeTypescriptOptons?: withAwesomeTypescript.AwesomeTypescriptOptions,
-  nextConfiguration?: NextConfiguration
-): NextConfiguration;
+declare function withAwesomeTypescript(nextOptions: {
+  awesomeTypescriptOptions?: withAwesomeTypescript.AwesomeTypescriptOptions,
+  webpack?(config: NextConfiguration, options: any): NextConfiguration
```
Comment (saitonakamura) [inline index.d.ts]: There will be an error (at least from some version of typescript) `Object literal may only specify known properties` if you provide something other than this two keys. And personally, I would refrain from making assumptions about `NextConfiguration`, Maybe it's better to rewrite it like this
```
declare function withAwesomeTypescript(
  nextConfig:
    & { awesomeTypescriptOptons?: withAwesomeTypescript.AwesomeTypescriptOptions }
    & NextConfiguration
)
```

**Review by saitonakamura — CHANGES_REQUESTED** [review]
Oh, that's a stupid bug, thanks for the pr. There's still one bug though.

File: index.d.ts (hunk 1 above)
Comment (Gavinchen92) [reply index.d.ts]: I just learned typescript. And i found the error in my project. please fix it, thank you

Comment (saitonakamura) [reply index.d.ts]: I commited a fix in your master branch, can you test it on your side?

**Comment by Gavinchen92** [comment]
close and open a new one

**Pull request closed without merging**